node/src/components/chain_synchronizer/error.rs

```diff
@@ -143,6 +143,16 @@ pub(crate) enum Error {
         #[serde(skip_serializing)]
         FetchTrieError,
     ),
+
+    /// A received trie matched the correct hash, but failed to serialize into a valid trie.
+    #[error("corrupt trie {hash}: {error}")]
+    CorruptTrie {
+        /// The hash of the corrupt trie.
+        hash: Digest,
+        #[serde(skip_serializing)]
```

Comment:
I suspect we are serializing our errors with JSON for logging purposes.  If so, why are we skipping the error message string?

Reply:
The serialization is not used in logging, but in queue dumps only (logs don't allow for arbitrary JSON to be stored).

The skip here is a mistake though, leftover from an intermediate state of the code.

Fixed in 635019f43